src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -4860,6 +4860,149 @@ class StubGenerator: public StubCodeGenerator {
+             tmp1   = r10, tmp2 = r11, tmp1h = rscratch1, tmp2h = rscratch2;
+
+
+    Label LOOP_COMPARE64, LOOP_COMPARE16, DIFF, LESS16, LESS8, CAL_DIFFERENCE, LENGTH_DIFF;
+
+    int largeLoopSize = 64/(isLL ? 1 : 2);
+
+    // before jumping to stub, pre-load 8 bytes already, so do comparison directly
+    __ eor(rscratch2, tmp1, tmp2);
+    __ cbnz(rscratch2, CAL_DIFFERENCE);
+
+    __ sub(cnt2, cnt2, wordSize/(isLL ? 1 : 2));
+    // update pointers, because of previous read
+    __ add(str1, str1, wordSize);
+    __ add(str2, str2, wordSize);
+
+    __ align(OptoLoopAlignment);
+    __ bind(LOOP_COMPARE64);
+      __ ldp(tmp1, tmp1h, Address(str1));
+      __ ldp(tmp2, tmp2h, Address(str2));
+      __ cmp(tmp1, tmp2);
+      __ ccmp(tmp1h, tmp2h, 0, Assembler::EQ);
+      __ br(__ NE, DIFF);
+
+      __ ldp(tmp1, tmp1h, Address(str1,16));
+      __ ldp(tmp2, tmp2h, Address(str2,16));
+      __ cmp(tmp1, tmp2);
+      __ ccmp(tmp1h, tmp2h, 0, Assembler::EQ);
+      __ br(__ NE, DIFF);
+
+      __ ldp(tmp1, tmp1h, Address(str1, 32));
+      __ ldp(tmp2, tmp2h, Address(str2, 32));
+      __ cmp(tmp1, tmp2);
+      __ ccmp(tmp1h, tmp2h, 0, Assembler::EQ);
+      __ br(__ NE, DIFF);
+
+      __ ldp(tmp1, tmp1h, Address(str1,48));
+      __ ldp(tmp2, tmp2h, Address(str2,48));
+      __ cmp(tmp1, tmp2);
+      __ ccmp(tmp1h, tmp2h, 0, Assembler::EQ);
+      __ br(__ NE, DIFF);
+
+      __ sub(cnt2, cnt2, isLL ? 64 : 32);
+      __ add(str1, str1, 64);
+      __ add(str2, str2, 64);
+      __ subs(rscratch2, cnt2, largeLoopSize);
+      __ br(__ GE, LOOP_COMPARE64);
+      __ cbz(cnt2, LENGTH_DIFF); // no more chars left?
+
+
+    __ subs(rscratch1, cnt2, isLL ? 16 : 8);
+    __ br(__ LE, LESS16);
+    __ align(OptoLoopAlignment);
+
+    __ bind(LOOP_COMPARE16);
+      __ ldp(tmp1, tmp1h, Address(__ post(str1, 16)));
+      __ ldp(tmp2, tmp2h, Address(__ post(str2, 16)));
+
+      __ cmp(tmp1, tmp2);
+      __ ccmp(tmp1h, tmp2h, 0, Assembler::EQ);
+      __ br(__ NE, DIFF);
+      __ sub(cnt2, cnt2, isLL ? 16 : 8);
+      __ subs(rscratch2, cnt2, isLL ? 16 : 8);
+      __ br(__ LT, LESS16);
+
+      __ ldp(tmp1, tmp1h, Address(__ post(str1, 16)));
+      __ ldp(tmp2, tmp2h, Address(__ post(str2, 16)));
+
+      __ cmp(tmp1, tmp2);
+      __ ccmp(tmp1h, tmp2h, 0, Assembler::EQ);
+      __ br(__ NE, DIFF);
+      __ sub(cnt2, cnt2, isLL ? 16 : 8);
+
+      __ subs(rscratch2, cnt2, isLL ? 16 : 8);
+      __ br(__ GE, LOOP_COMPARE16);
+      __ cbz(cnt2, LENGTH_DIFF);
+
+    __ bind(LESS16);
+      // each 8 compare
+      __ subs(cnt2, cnt2, isLL ? 8 : 4);
+      __ br(__ LE, LESS8);
+      __ ldr(tmp1, Address(__ post(str1, 8)));
+      __ ldr(tmp2, Address(__ post(str2, 8)));
+      __ eor(rscratch2, tmp1, tmp2);
+      __ cbnz(rscratch2, CAL_DIFFERENCE);
+      __ sub(cnt2, cnt2, isLL ? 8 : 4);
+
+    __ bind(LESS8);                          // directly load last 8 bytes
+    if(!isLL) {
```

Comment:
The indentation of the `if` seems wrong here: shouldn't it line up with the assembly block like the `if` on line 4989? Also needs a space before the parenthesis.